feat: list of venues

KindImagination opened the pull request:
closes #218 

How to test:
Open Venues, you can see the first 9 venues. 
you can load another 9 venues by clicking the "load more venues" button. 
you can filter by "dance style", "location" and "name".


ISSUE: if you flter by Locations: Munich - Germany 

one of the vanue hase name: "64 Rue Orfila" and Adress: "64 Rue Orfila, 75020 Paris, France"

this is Backup Error, this Venue hase City: "Munich"

KindImagination committed:
feat: optimize venue fetching and display in VenuesView

- Implemented server-side filtering and selective field retrieval for venues.
- Introduced a two-phase query approach to fetch basic venue data and styles separately.
- Limited event retrieval to enhance performance and added logging for performance measurement.
- Increased pagination size from 9 to 12 to reduce the number of requests.
- Updated venue card display to handle missing properties safely and optimized image loading with NuxtImg.
- Added loading states and a "Load More" button for better user experience.

KindImagination committed:
refactor: clean up comments and remove unused code in VenuesView

KindImagination committed:
feat: implement caching for dance styles in VenuesView

- Added caching mechanism for dance styles using localStorage to improve performance and reduce API calls.
- Refactored fetchStyles function to load styles from cache if available, enhancing user experience.
- Updated pagination limit back to 9 for consistency.
- Introduced loading state for dance styles while fetching data.
- Modified button functionality to toggle filters and fetch styles accordingly.

KindImagination committed:
feat: enhance dance style selection and display in VenuesView

- Implemented debounce functionality for search input, optimizing performance during style selection.
- Added toggle functionality for selecting and deselecting dance styles, with immediate venue refresh.
- Introduced a clear all button for selected styles, improving user experience.
- Enhanced display of dance styles with counts, providing better visibility of available options.
- Updated UI elements for improved layout and user interaction.

KindImagination committed:
refactor: remove unused features and improve UI elements in VenuesView

- Eliminated commented-out code related to features, price range, and capacity filters to clean up the component.
- Updated UI to indicate that certain features and filters are "Coming soon," enhancing user clarity.
- Improved the layout and interaction of the features and price range sections for better user experience.

KindImagination committed:
refactor: clean up comments and remove unused code in profiles router

- Removed commented-out code related to dance styles extraction and event combination, enhancing code clarity and maintainability.

KindImagination committed:
feat: implement dynamic location fetching in LocationPanel

- Added functionality to fetch unique cities and regions from the server, enhancing the accuracy of location data.
- Introduced loading states for cities and regions to improve user experience during data retrieval.
- Implemented fallback default locations in case of fetch errors, ensuring consistent UI behavior.
- Updated computed properties to work with reactive data structures for better performance and reactivity.

KindImagination committed:
refactor: clean up comments

razbakov commented on components/VenuesView.vue:
```diff
@@ -1,78 +1,213 @@
+const loadStylesFromCache = () => {
+  if (process.client) {
+    try {
+      const cached = localStorage.getItem(CACHE_KEY)
```
this is bad. better just use vue-query

KindImagination replied:
solved

razbakov commented on components/VenuesView.vue:
```diff
@@ -128,156 +266,230 @@ const clearLocationFilter = () => {
-            {{ Math.min(...venue.areas.map((a) => a.pricePerHour)) }}-{{
-              Math.max(...venue.areas.map((a) => a.pricePerHour))
+            {{
+              Math.min(
```
so that would be the minimum of what? of shown venues? this have to be done across all venues and you don't have that information on client, so you need to do it in backend

KindImagination replied:
This part is not in the database

KindImagination committed:
refactor: remove caching logic and simplify dance styles fetching in VenuesView

- Eliminated caching mechanism for dance styles, streamlining the data fetching process.
- Updated fetchStyles function to directly retrieve styles from the server without caching.
- Enhanced the UI message for no venues found to provide clearer feedback based on selected styles.

KindImagination committed:
refactor: remove unused bio search filter in profiles router

- Eliminated the commented-out bio search filter from the profiles router, enhancing code clarity and maintainability.

KindImagination committed:
refactor: remove price and capacity display logic in VenuesView

- Eliminated the price per hour and capacity display logic for venue areas to simplify the component.
- Added a placeholder comment for future implementation of price and capacity features.
- Updated the display to show venue bio if available, enhancing the information presented to users.

razbakov commented on components/LocationPanel.vue:
```diff
@@ -11,29 +11,62 @@ const props = defineProps({
+  return cities.value.filter((city) => city.name.toLowerCase().includes(search))
+})
+
+onMounted(() => {
```
that should be async

razbakov commented on components/LocationPanel.vue:
```diff
@@ -66,22 +112,45 @@ const filteredCities = computed(() => {
 </template>
+
+<style scoped>
+.city-list {
```
use tailwind

razbakov commented on components/VenuesView.vue:
```diff
@@ -1,78 +1,170 @@
 
-const allFeatures = computed(() =>
-  Array.from(new Set(venues.flatMap((venue) => venue.features)))
+const allStyles = computed(() =>
```
list should come from backend

KindImagination committed:
refactor: update fetchLocations call in LocationPanel to be asynchronous

- Modified the onMounted lifecycle hook to await the fetchLocations function, ensuring proper handling of asynchronous data retrieval.
- This change enhances the reliability of location data fetching during component initialization.

KindImagination committed:
refactor: use tailwind

KindImagination committed:
refactor: simplify dance styles computation in VenuesView

KindImagination committed:
refactor: enhance search filtering logic in profiles router

- Updated the search filtering mechanism to use AND conditions for name and location queries, improving the accuracy of search results.
- Consolidated the location search logic to streamline the query structure and enhance code maintainability.

KindImagination committed:
refactor: enhance location data structure in LocationPanel

- Updated the cities and regions data structure to include a 'type' property, improving clarity and categorization of location data.
- Adjusted the fetchLocations function to populate the new 'type' field for cities and regions.
- Modified computed properties and template bindings to utilize the updated data structure, ensuring accurate rendering of location information.

KindImagination committed:
refactor: enhance city data structure and filtering in LocationPanel

- Updated the cities data structure to include additional properties such as displayName, country, and countryCode for improved clarity and usability.
- Modified the fetchLocations function to populate the new fields and removed the regions data handling, as it is no longer utilized.
- Enhanced the filtering logic for cities to include displayName and country in the search criteria, improving the user experience when searching for locations.
- Adjusted the template bindings to reflect the updated data structure, ensuring accurate rendering of city information.

KindImagination committed:
refactor: add search functionality for dance styles in VenuesView

- Introduced a new search input for filtering dance styles based on user input, enhancing the user experience.
- Implemented a computed property to filter dance styles dynamically, ensuring only relevant styles are displayed.
- Updated the template to reflect the new search feature and improved the layout for better usability.
- Renamed "Features" section to "Amenities" for clarity.

KindImagination committed:
refactor: update venue links in VenuesView

razbakov commented on components/LocationPanel.vue:
```diff
@@ -11,29 +11,85 @@ const props = defineProps({
+
+    cities.value = [
+      {
+        name: 'New York',
```
what's this for?

razbakov commented on components/VenuesView.vue:
```diff
@@ -128,176 +231,199 @@ const clearLocationFilter = () => {
-            people
+          <div class="flex items-center gap-1 text-xs text-muted-foreground">
+            <Icon name="ph:calendar-check" class="w-3 h-3" />
+            <span>Dance styles from events</span>
```
what's this for?

KindImagination replied:
removed

KindImagination committed:
refactor: implement searchable dropdown for dance styles in VenuesView

- Added a new searchable dropdown component for selecting dance styles, improving user interaction.
- Introduced state management for selected dance styles and search queries.
- Enhanced the template to support multi-selection of dance styles with visual feedback.
- Updated the logic to keep the dropdown open for additional selections, streamlining the user experience.

KindImagination committed:
refactor: remove redundant dance styles label in VenuesView

- Removed the label displaying "Dance styles from events" to streamline the component layout.
- This change enhances the visual clarity of the dance styles section by reducing unnecessary text.

KindImagination committed:
refactor: streamline city data handling in LocationPanel and remove filters section in VenuesView

- Cleared the cities array in LocationPanel to handle errors more gracefully when fetching locations.
- Removed the filters section from VenuesView to simplify the component layout and improve user experience.
